app/streaming/video/ffmpeg-renderers/d3d11va.h

```diff
@@ -30,24 +34,99 @@ class D3D11VARenderer : public IFFmpegRenderer
     bool setupVideoTexture();
     void renderOverlay(Overlay::OverlayType type);
     void bindColorConversion(AVFrame* frame);
+    void prepareVideoProcessorStream(AVFrame* frame);
     void renderVideo(AVFrame* frame);
+    bool createVideoProcessor(bool reset = false);
+    void enableAMDVideoSuperResolution(bool activate = true);
+    void enableIntelVideoSuperResolution(bool activate = true);
+    void enableNvidiaVideoSuperResolution(bool activate = true);
+    void enableAMDHDR(bool activate = true);
+    void enableIntelHDR(bool activate = true);
+    void enableNvidiaHDR(bool activate = true);
     bool checkDecoderSupport(IDXGIAdapter* adapter);
     bool createDeviceByAdapterIndex(int adapterIndex, bool* adapterNotFound = nullptr);
+    void setHDRStream();
+    void setHDROutPut();
 
     int m_DecoderSelectionPass;
 
+    int m_AdapterIndex = 0;
+    int m_OutputIndex = 0;
     IDXGIFactory5* m_Factory;
     ID3D11Device* m_Device;
     IDXGISwapChain4* m_SwapChain;
     ID3D11DeviceContext* m_DeviceContext;
     ID3D11RenderTargetView* m_RenderTargetView;
     SDL_mutex* m_ContextLock;
 
+    ID3D11VideoDevice* m_VideoDevice;
+    ID3D11VideoContext2* m_VideoContext;
+    CComPtr<ID3D11VideoProcessor> m_VideoProcessor;
+    CComPtr<ID3D11VideoProcessorEnumerator> m_VideoProcessorEnumerator;
+    D3D11_VIDEO_PROCESSOR_OUTPUT_VIEW_DESC m_OutputViewDesc;
+    D3D11_VIDEO_PROCESSOR_INPUT_VIEW_DESC m_InputViewDesc;
+    D3D11_VIDEO_PROCESSOR_STREAM m_StreamData;
+    Microsoft::WRL::ComPtr<ID3D11VideoProcessorOutputView> m_OutputView;
+    Microsoft::WRL::ComPtr<ID3D11VideoProcessorInputView> m_InputView;
```

Comment:
Why different smart pointer classes? (`CComPtr` and `Microsoft::WRL::ComPtr`)

Reply:
Done
There were no specific reason, I was just able to get it work with CComPtr but not with Microsoft::WRL::ComPtr, but it was because I miss used the pointer, now it is done corectly and I could make the switch.